src/NumericalAlgorithms/LinearOperators/Transpose.hpp

```diff
@@ -20,13 +20,15 @@
 /// another function can operate on contiguous chunks of memory.
 ///
 /// \requires `transpose.size()` to be the product of `number_of_chunks` and
-/// `chunk_size, `u.size()` to be equal or greater than `transpose.size()`,
+/// `chunk_size`, `u.size()` to be equal or greater than `transpose.size()`,
 /// and that both `transpose` and `u` have a `data()` member function.
 ///
-/// \details The container `u` holds a contiguous array of data that is at least
-/// of size `number_of_chunks` times `chunk_size`.  The output `transpose` has
-/// its data arranged such that the first `number_of_chunks` elements in
-/// `transpose` will be the first element of each chunk of `u`.  The last
+/// \details The container `u` holds a contiguous array of data, the
+/// initial part (commonly all) of which is treated as a sequence of
```

Comment:
Personally, I'd find it clearer if the "partial transpose" feature were not mentioned until after I had read about how the transposing is done. To me, the ability to partially transpose the data seems like a higher-order detail.

So I'd suggest changing the first sentence to:
"The container `u` holds a contiguous array of data, treated as a sequence of..."

And adding a new sentence at the end, roughly like this:
"If `u.size()` is greater than `transpose.size()`, then `u` is partially transposed into `transpose`: the first `transpose.size()` elements of `u` are transposed as previously described, but subsequent elements of `u` do not appear in `transpose`."

If you disagree with this suggestion, feel free to disregard.

Reply:
Reworded along those lines.